crates/pallet-domains/src/staking.rs

```diff
@@ -0,0 +1,366 @@
+//! Staking for domains
+
+use crate::pallet::{
+    DomainStakingSummary, NextOperatorId, OperatorIdOwner, OperatorPools, PendingTransfers,
+};
+use crate::{BalanceOf, Config, FreezeIdentifier};
+use codec::{Decode, Encode};
+use frame_support::traits::fungible::{Inspect, InspectFreeze, MutateFreeze};
+use frame_support::traits::tokens::{Fortitude, Preservation};
+use frame_support::{ensure, PalletError};
+use scale_info::TypeInfo;
+use sp_core::Get;
+use sp_domains::{DomainId, EpochIndex, ExecutorPublicKey, OperatorId};
+use sp_runtime::traits::{CheckedAdd, Zero};
+use sp_runtime::Percent;
+use sp_std::vec::Vec;
+
+/// Type that represents an operator pool details.
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct OperatorPool<Balance> {
+    pub signing_key: ExecutorPublicKey,
+    pub current_domain_id: DomainId,
+    pub minimum_nominator_stake: Balance,
+    pub nomination_tax: Percent,
+    /// Total active stake for the current pool.
+    pub current_total_stake: Balance,
+    /// Total stake for the current pool in the next epoch.
+    pub next_total_stake: Balance,
+    /// Total shares of the nominators and the operator in this pool.
+    pub total_shares: Balance,
```

Comment:
Using `Balance` here could imply that the `_stake` balances are of the same unit.

Reply:
that is fine since we need a type that is <> with Balance, so using this should not have any issues